src/utils/MessageBuilder.ts

```diff
@@ -211,6 +272,20 @@ export default class MessageBuilder {
 
     const orangeTeam: Array<string> = [];
     const blueTeam: Array<string> = [];
+    const mmrBlue = await calculateMMR(ballchasers[0].id, Team.Blue);
```

Comment:
What do you think about moving some of this async logic to the controller and out of MessageBuilder and then just passing it in? 🤔

Reply:
I feel like if it were moved, it might clean up the size of activeMatchMessage, but wouldn't make much difference. Since MessageBuilder is a util, does the logic need to move to the controller? This is the only time and place some of this logic needs to be done like `if (mmrBlue < mmrOrange) {` all the way through the `else`. 
I suppose either way can work.

Reply:
Right, MessageBuilder is a util so it feels weird to have business logic inside of a util like this. Would it make more sense for the TeamAssignment service to return all the info needed for the embed? So it would return the teams, mmr stake, and win probability. That way the business logic still lives in the service layer and you can just forward it into the util.

Thoughts on this approach?

Reply:
I'm alright with this approach. I'll make a new function there that contains all of the logic to return.

Reply:
Should this also include the for loop containing the players per team that will be pushed to the embed? As setup for broken queue?

Reply:
Yea so I think the interface for what to return from TeamAssignmentService would look something like:
```ts
interface ActiveMatchTeamDetails {
   players: ReadonlyArray<PlayerInActiveMatch>;
   mmrStake: number;
   winProbability;
}

// This being the interface for the object that is returned.
interface ActiveMatchCreated {
   orange: ActiveMatchTeamDetails;
   blue: ActiveMatchTeamDetails;
};
```
When we get to broken queue we can add a new array to the `ActiveMatchCreated` interface with a list of players who have voted to broken queue.